handlers/routeservice.go

```diff
@@ -62,6 +62,33 @@ func (r *RouteService) ServeHTTP(rw http.ResponseWriter, req *http.Request, next
 	routeServiceURL := reqInfo.RoutePool.RouteServiceUrl()
 	if routeServiceURL == "" {
 		// No route service is associated with this request
+		if r.config.StrictSignatureValidation() && len(req.Header.Values(routeservice.HeaderKeySignature)) > 0 {
+			// Someone is still setting the header, possibly trying to impersonate a route service
+			// request. We will validate the request to ensure the header is valid.
+			// TODO: what do we have to provide as app URL?
+			rreq := newRequestReceivedFromRouteService("", req.Header)
+			_, err := r.config.ValidateRequest(rreq)
+			if err != nil {
+				AddRouterErrorHeader(rw, "invalid_route_service_signature")
+				if errors.Is(err, routeservice.ErrExpired) {
+					r.errorWriter.WriteError(
+						rw,
+						http.StatusGatewayTimeout,
+						fmt.Sprintf("Failed to validate Route Service Signature: %s", err.Error()),
```

Comment:
For both error messages, maybe we can add the target URL? Sth like this (untested and did not check Sprintf syntax):
```suggestion
						fmt.Sprintf("Failed to validate Route Service Signature to %s: %s", req.URL.String(), err.Error()),
```

It helps finding who is running into this without correlating trace-IDs.

Reply:
This string is also what is written in the response that is sent to the client. Given that the access log can be correlated via some ID I'd rather not make the string more verbose.

Reply:
Shouldn't we split the client response and our error log then? I cannot judge the effort and just know the pain when you want to look at multiple data-points but can only get data via the correlation. I am fine keeping it as is, if the effort it too high.

Reply:
I think this is more of a general issue with the error reporting around route-services. I will not address it with this pull request as it seems complex, but we should look into this as I also feel the pain of correlating between requests, route-services and error logs.

Reply:
@plowin I think the x_cf_routererror in gorouter access_log is very helpful:
`x_cf_routererror:"invalid_route_service_signature"`. The only thing we cannot distinguish between expired signature and invalid in access log.

Reply:
We can introduce two different `x_cf_routererror`.
- for expired: `x_cf_routererror:"expired_route_service_signature"`
- for other errors: `x_cf_routererror:"invalid_route_service_signature"`

Reply:
I've included that.

Reply:
Thx for the hints and clarifying! Then I am totally fine and was still stuck to the very initial version, resolving.